### DISCUSSION TITLE
q-select kees querying server while filtering

### DISCUSSION THREAD

**Discussion by aarzuagat** [quasarframework/quasar#10556] [General - Components / Directives / etc] [answered]
Hi guys!
I'm using q-select to filter cities. More than 44 millions of cities are in database, so I query it to filter. But, this is the damn issue. I'm watching Network tab in browser and.... surprise! Many queries are done to server, even my word has 4 letters. 
For example, for "barcelona", I got this:
![imagen](https://user-images.githubusercontent.com/31313488/131734058-7ab174f9-3aad-4b2f-9e3e-ca40fd4ff7c6.png)

Is there a way to force stopping querying function?
This is my filter function
![imagen](https://user-images.githubusercontent.com/31313488/131734422-cdb24f23-8435-4f71-bb3d-9b932e3cb368.png)

**Comment by metalsadman** [comment, marked as answer]
@aarzuagat call update inside your api call (at `then` etc..) not the other way around, there's also a third argument for filter prop `abort` you can use when no result was found maybe on your `catch` or `finally` callbacks. see example https://quasar.dev/vue-components/select#example--lazy-filtering
